### PR TITLE
CORDA-2521: Checkpoint verifier no longer cares about the CorDapp jar name

### PR DESCRIPTION
**Pull request by shamsasari**
The check on the CorDapp hash is sufficient.

### CODE REVIEW

File: node/src/main/kotlin/net/corda/node/internal/CheckpointVerifier.kt [hunk 1]
```diff
@@ -18,42 +18,57 @@ object CheckpointVerifier {
...
-            when (installedCordapps.size) {
-                0 -> throw CheckpointIncompatibleException.FlowNotInstalledException(subFlow.flowClass)
-                1 -> {
-                    val currenCordapp = installedCordapps.first()
-                    if (corDappInfo.corDappHash != currenCordapp.jarHash) {
-                        throw CheckpointIncompatibleException.FlowVersionIncompatibleException(subFlow.flowClass, currenCordapp, corDappInfo.corDappHash)
-                    }
-                }
-                else -> throw IllegalStateException("Multiple Cordapps with name ${corDappInfo.corDappName} installed.") // This should not happen
+        if (subFlowVersion is SubFlowVersion.CorDappFlow && subFlowVersion.corDappHash !in currentCordapps) {
```
Comment (tudor-malene) [inline node/src/main/kotlin/net/corda/node/internal/CheckpointVerifier.kt]: This logic seems correct but is really hard to follow.

I think a comment describing how it works would help.

**Comment by shamsasari** [comment]
Comments added. Back to you.
